src/nfa/nfa.cc

```diff
@@ -555,3 +555,36 @@ bool Nfa::is_identical(const Nfa& aut) const {
     }
     return delta == aut.delta;
 }
+
+Nfa& Nfa::unite_nondet_with(const mata::nfa::Nfa& aut) {
+    const size_t num_of_states{ this->num_of_states() };
+    const size_t aut_num_of_states{ aut.num_of_states() };
+    const size_t new_num_of_states{ num_of_states + aut_num_of_states };
+    auto renumber_states = [&](State st) {
+        return st + num_of_states;
+    };
+
+    if (this == &aut) {
+        throw std::runtime_error("Performing a useless in-place union on the same NFA object");
```

Comment:
Should this be error? We can just return `this`.

Reply:
That sounds good to me. I only copied the code from the original implementation without modifications, but I think this is a reasonable modification.